template-processors/openshift-provision/files/filter_plugins/cpu_cores_filter.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2019 Kohl's Department Stores, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#pylint: disable=invalid-name
+import re
+
+def number_of_cores(source, node_type):
+    ''' Returns number of machine type cpu cores '''
+    machine_type = source[node_type]["gcp"]["machine_type"]
+    cores = re.search(r"-(?P<cores>\d{1,2})(-.{3,})?$", machine_type)
+    if cores:
+        return cores.group("cores")
+    else:
+        raise ValueError("number_of_cores filter failed while \
+        trying to find number of cores for {0} node group".format(node_type))
+
+def preemptible_pod_cpu_request(source, node_type, reduction=1):
+    ''' Calculates preemptible pods cpu request as a number of cores a machine has
+        reduced by default by 1 (to provide space for DaemonSets) '''
+    return int(number_of_cores(source, node_type)) - reduction
+
+class FilterModule(object):
+    ''' OpenShift Filters for selector strings '''
+
+    # pylint: disable=no-self-use, too-few-public-methods
+    def filters(self):
+        ''' Returns filters provided by this class '''
+        return {
+            'number_of_cores': number_of_cores,
+            'preemptible_pod_cpu_request': preemptible_pod_cpu_request
+        }
```

Comment:
What is this file? Why is it here?

I guess those are probably some tools for use in the templates? If yes, I'd strongly recommend adding them with a separate subsequent commit, with an explanation in commit message. The base of this commit doesn't seem to require them to work in a minimal capacity, IIUC, and it's huge enough already as is, without the extra .py files.

Reply:
Good call, I'll pull these out for now and add them in if necessary in a future PR. Sorry for the distracting code!